wagtail/admin/api/actions/publish.py

```diff
@@ -11,37 +10,13 @@
 from .base import APIAction
 
 
-class PublishPageAPIActionSerializer(Serializer):
-    changed = fields.BooleanField(default=True, required=False)
-    log_action = fields.BooleanField(default=True, required=False)
-    previous_revision_id = fields.IntegerField(required=False)
-
-
 class PublishPageAPIAction(APIAction):
-    serializer = PublishPageAPIActionSerializer
+    serializer = Serializer
 
     def _action_from_data(self, instance, data):
-        previous_revision = None
         user = self.request.user
-        log_action = data["log_action"]
-
-        previous_revision_id = data.get("previous_revision_id", None)
-        if previous_revision_id:
-            previous_revision = get_object_or_404(
-                instance.revisions, id=previous_revision_id
-            )
-
-        revision = instance.save_revision(
-            user=user, log_action=log_action, previous_revision=previous_revision
-        )
-
-        return PublishPageRevisionAction(
-            revision,
-            user=user,
-            changed=data["changed"],
-            log_action=log_action,
-            previous_revision=previous_revision,
-        )
+        revision = instance.get_latest_revision() or instance.save_revision(user=user)
```

Comment:
I think the ``or`` operator will always execute both sides of the operator regardless of the value of the first one.

We should change this to:
revision = instance.get_latest_revision()
if revision is None:
    revision = instance.save_revision(user=user)

Reply:
Actually, I might be wrong here:

```python
>>> print("foo") and print("bar")
foo
>>> print("foo") or print("bar")
foo
bar
```